src/main/java/com/google/firebase/remoteconfig/RemoteConfigParameter.java

```diff
@@ -22,51 +22,99 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * Represents a Remote Config parameter that can be included in a {@link RemoteConfigTemplate}.
+ * At minimum, a default value or a conditional value must be present for the
+ * parameter to have any effect.
+ */
 public final class RemoteConfigParameter {
 
   private RemoteConfigParameterValue defaultValue;
   private String description;
   private Map<String, RemoteConfigParameterValue> conditionalValues;
 
+  /**
+   * Creates a new {@link RemoteConfigParameter}.
+   */
   public RemoteConfigParameter() {
     conditionalValues = new HashMap<>();
-    defaultValue = ExplicitParameterValue.of("");
+    defaultValue = RemoteConfigParameterValue.of("");
```

Comment:
I feel like both these should remain null until explicitly set. It's ok to let the backend choose default values for these as necessary.

Reply:
I see your point. Updated the code and added `null` checks in conversions.
I have a few concerns with setting these to `null` though. Isn't it recommended not to return `null` collections in Java? If we don't set to `conditionalValues` to an empty map at init, the following will throw an exception on the client side.
```
RemoteConfigParameter p = new RemoteConfigParameter().getConditionalValues().put("ios_en", ParameterValue.of("text")```

This can be prevented with a simple `null` check on the client side, though.